In [1]:
import tensorflow as tf
from tensorflow.keras import datasets, layers, models, preprocessing 

In [2]:
max_len = 200
n_words = 10000
dim_embedding = 256 
EPOCHS = 20 
BATCH_SIZE = 500
DROPOUT1 = 0.3
DROPOUT2 = 0.5

In [3]:
def load_data():
  (X_train, y_train), (X_test, Y_test) = datasets.imdb.load_data(num_words=n_words)
  X_train = preprocessing.sequence.pad_sequences(X_train,maxlen=max_len)
  X_test = preprocessing.sequence.pad_sequences(X_test,maxlen=max_len)
  return (X_train, y_train), (X_test, Y_test)

def build_model():
  model = models.Sequential()
  model.add(layers.Embedding(n_words, dim_embedding, input_length=max_len))
  model.add(layers.Dropout(DROPOUT1))
  model.add(layers.GlobalMaxPooling1D())
  model.add(layers.Dense(128, activation='relu'))
  model.add(layers.Dropout(DROPOUT2))
  model.add(layers.Dense(1, activation='sigmoid'))
  return model

In [4]:
(X_train, y_train), (X_test, y_test) = load_data()
model = build_model()

In [5]:
model.summary()

model.compile(optimizer='Adam',loss='binary_crossentropy',metrics=['accuracy'])

score = model.fit(X_train, y_train,epochs=EPOCHS, batch_size=BATCH_SIZE, validation_data=(X_test,y_test))


Model: "sequential"
_________________________________________________________________
Layer (type)                 Output Shape              Param #   
embedding (Embedding)        (None, 200, 256)          2560000   
_________________________________________________________________
dropout (Dropout)            (None, 200, 256)          0         
_________________________________________________________________
global_max_pooling1d (Global (None, 256)               0         
_________________________________________________________________
dense (Dense)                (None, 128)               32896     
_________________________________________________________________
dropout_1 (Dropout)          (None, 128)               0         
_________________________________________________________________
dense_1 (Dense)              (None, 1)                 129       
Total params: 2,593,025
Trainable params: 2,593,025
Non-trainable params: 0
______________________________________________

In [6]:
score = model.evaluate(X_test, y_test, batch_size=BATCH_SIZE)
print("\nTest score:", score[0])
print("Test accuracy:", score[1])

50/50 [==============================] - 1s 22ms/step - loss: 0.4933 - accuracy: 0.8508

Test score: 0.493252694606781
Test accuracy: 0.8507599830627441


I am not happy at all with my current results -- 85%. I am going to have to keep tweaking to get better results.